### ISSUE TITLE
QT icon not showing after updating to plasma 6

### ISSUE THREAD

**Issue by MarianHabor** [enielrodriguez/optimus-gpu-switcher#28]
QT icon trail not showing after updating to plasma 6

**Comment by elman22** [comment]
Yes. It's not compatible with Plasma 6. Could you please fix that? Thanks a lot :smiley:

**Comment by r4-broadcast** [comment]
Will there be compatibility with Plasma 6?

**Comment by MarianHabor** [comment]
This needs porting to plasma 6 [Porting Plasmoids to KF6](https://develop.kde.org/docs/plasma/widget/porting_kf6/)

**Comment by enielrodriguez** [comment]
Thanks to [@LTSlw](https://github.com/LTSlw) for the contribution. Optimus GPU Switcher 1.0.0 for KDE 6 is now in the KDE Store.

**Comment by andreaemonti** [comment]
I uninstalled the previous plasmoid, then I installed the version 1.0.0 for Plasma 6 from the "add more objects" store.
The tray icon shows the correct mode I'm in, but doesn't open the dialog on click. I can only right-click it and open its settings, but left-clicking the icon doesn't do anything.

System (up to date) running Endeavour OS + Plasma 6 + Wayland

**Comment by Labaman** [comment]
I'm seeing a similar problem.
But I did some more tests, and came to the conclusion that apparently KDE6 uses a new notification method for system tray applications. 
I added an additional widget instance, but moved it outside the system tray. As a result, the popup notifications for the instance outside the system tray displayed correctly, while the first instance did not display left-click popup notifications.  I also can't remove the first instance of the widget from the tray while in edit mode. 
![Optimus](https://github.com/enielrodriguez/optimus-gpu-switcher/assets/30574992/76760efe-96b0-4806-afd9-d89f2d3f8f2b)
 
Thus, it is necessary to either change the widget type, unlinking it from the system tray, or rework the methods used to handle notifications.

**Comment by andreaemonti** [comment]
Ah, yes, sorry. I didn't notice this. The plasmoid itself works perfectly. I can add it to the panel wherever, and works on click.
The icon in the system tray is the one provided by the notification centre. That's the one not responding on click.

>  I also can't remove the first instance of the widget from the tray while in edit mode.

You cannot edit those in panel edit mode, since they belong all together to the system tray. But you can customise which icons to show/hide in the arrow-up-menu/hide completely in the system tray settings (cog icon from arrow-up-menu).

**Comment by Labaman** [comment]
Yeah, that's what I did. But it's more like a workaround than a solution to the problem. 
By the way, if you exclude the panel from the system tray (change `"EnabledByDefault": true `-> false in `metadata.json`, the problem is similar - the first instance of plasmoid does not respond to left mouse button click.

**Comment by LTSlw** [comment]
It's a bug. I didn't notice that because I use the second type. I'll fix it soon

**Comment by enielrodriguez** [comment]
Published in KDE Store! Thanks @LTSlw !
